spec/lib/alpha_taxonomy/taxon_linker_spec.rb

```diff
@@ -0,0 +1,42 @@
+require "rails_helper"
+
+RSpec.describe AlphaTaxonomy::TaxonLinker do
+  describe "#run!" do
+    def stub_taxon_fetch(title:, content_id:)
+      allow(Services.content_store).to receive(:content_item!).with(
+        AlphaTaxonomy::TaxonPresenter.new(title: title).base_path
+      ).and_return("content_id" => content_id)
+    end
+
+    def stub_content_item_fetch(base_path:, content_id:)
+      allow(Services.content_store).to receive(:content_item!).with(
+        base_path
+      ).and_return("content_id" => content_id)
+    end
+
+    it "creates taxon links based on the grouped mappings" do
+      import_file = AlphaTaxonomy::ImportFile.new
+      allow(import_file).to receive(:grouped_mappings).and_return(
+        "/a-foo-content-item" => ["Foo Taxon"],
+        "/a-foo-bar-content-item" => ["Foo Taxon", "Bar Taxon"]
+      )
+      allow(AlphaTaxonomy::ImportFile).to receive(:new).and_return(import_file)
```

Comment:
A decent option, but it makes running this command more burdensome. You can't just chuck the `run!` invocation into the rails `runner`, you'd either have to load up a console in order to instantiate the `import_file`, or wrap the whole thing in a rake task. Added to that is that the `TaxonLinker` isn't used in multiple contexts, making the use of dependency injection here a slightly odd choice.

Not sure it's worth the trade off, particularly as the readability improvement is slight at the most.